File: java-samples/src/main/java/com/jetbrains/debugging/SetBreakpointFromStackTrace.java
package com.jetbrains.debugging;

public class SetBreakpointFromStackTrace {
    // run this to see a "Create Breakpoint" hint in the stace trace in the run window
    public static void main(String[] args) {
        shouldThrowClassCastException();
    }

    //This is supposed to error
    private static void shouldThrowClassCastException() {
        int value = (int) getAString();
        System.out.println("value = " + value);
    }

    //<editor-fold desc="Helper methods">
    private static Object getAString() {
        return "a string";
    }
    //</editor-fold>
}

File: kotlin-samples/src/main/java/com/jetbrains/debugging/SetBreakpointFromStackTrace.kt
package com.jetbrains.debugging

//This is supposed to error
// when you run this you'll get an exception, with a hint to "Create breakpoint"
// this will let you set an Exception breakpoint on ArrayIndexOutOfBounds exceptions
fun main() {
    val ints = intArrayOf(35, 543, 56622)
    for (i in 1..10) {
        val anInt = ints[i]
        println("anInt = $anInt")
    }
}

